src/NeinLinq/SelectorTranslator.cs

```diff
@@ -39,16 +39,26 @@ public static class SelectorTranslator
             var leftInit = left.Body as MemberInitExpression;
             var rightInit = right.Body as MemberInitExpression;
 
-            if (leftInit == null || rightInit == null)
-                throw new NotSupportedException("Only member init expressions are supported yet.");
-            if (leftInit.NewExpression.Arguments.Any() || rightInit.NewExpression.Arguments.Any())
-                throw new NotSupportedException("Only parameterless constructors are supported yet.");
+            var leftNew = left.Body as NewExpression ?? leftInit?.NewExpression;
+            var rightNew = right.Body as NewExpression ?? rightInit?.NewExpression;
+
+            var bindings = Enumerable.Empty<MemberBinding>();
+
+            if (leftNew != null && rightNew != null)
+            {
+                if (leftNew.Arguments.Any() || rightNew.Arguments.Any())
+                    throw new NotSupportedException("Only parameterless constructors are supported yet.");
+
+                if (leftInit != null && rightInit != null)
+                    bindings = leftInit.Bindings.Concat(rightInit.Bindings);
```

Comment:
Good point! Unfortunately I didn't look at your solution and solved it based on the version I created - doesn't look that elegant, but it works. You might want to have a look if the coding style fits your needs though.